Tests/SwiftFusionTests/MCMC/MCMCTests.swift

```diff
@@ -15,16 +15,16 @@ class MCMCTests: XCTestCase {
   /// Sampling from the Standard Normal Distribution.
   /// Inspired by testRWM1DUniform from tfp
   func testRWM1DUniform() throws {
-    var deterministicEntropy = ARC4RandomNumberGenerator(seed: 42)
+    let deterministicEntropy = ARC4RandomNumberGenerator(seed: 42)
     
     // Create kernel for MCMC sampler:
     // target_log_prob_fn is proportional to log p(x), where p is a zero-mean Gaussian
     // new_state_fn is a symmetric (required for Metropolis) proposal density,
     // in this case perturbing x with uniformly sampled perturbations.
     let kernel = RandomWalkMetropolis(
       sourceOfEntropy: deterministicEntropy,
-      target_log_prob_fn: {(x:Double) in -0.5*x*x}, //  tfd.Normal(loc=dtype(0), scale=dtype(1))
-      new_state_fn: {(x:Double) in x + Double.random(in: -1..<1, using: &deterministicEntropy)}
+      target_log_prob_fn: {(x: Double) in -0.5*x*x}, //  tfd.Normal(loc=dtype(0), scale=dtype(1))
+      new_state_fn: {(x: Double, r: inout AnyRandomNumberGenerator) in x + Double.random(in: -1..<1, using: &r)}
```

Comment:
@dabrahams Do you think this is a good pattern?

Reply:
Well, that won't work without changing the Metropolis sampler. 

But, I am not convinced this is the problem: I obtained the histograms above by adding a print statement inside `new_state_fn`, and printing out the 2500 random numbers generated. So, it's definitely not the case that I got the same number every time...

Reply:
You get the same number *twice*, IMHO, so the distribution looks exactly the same

Reply:
> You get the same number _twice_, IMHO, so the distribution looks exactly the same

I don't understand

Reply:
@ProfFan I'm not sure it's a great pattern, because [here](https://github.com/borglab/SwiftFusion/blob/master/Sources/SwiftFusion/MCMC/RandomWalkMetropolis.swift#L34) the instance of `RandomWalkMetropolis` is storing its source of entropy (after wastefully wrapping it in another `AnyRandomNumberGenerator`).  Therefore, the two instances of `RandomWalkMetropolis` will see the same sequence of random numbers, but shifted.  I guess that's the case with all PRNGs, but in this case the shift will be by the same number of samples each time the test is run.  Do you think that's alright?  It might be the case that we should create a class wrapper to give `ARC4RandomNumberGenerator` reference semantics.

Reply:
I take it all back (except for the wasteful wrapping on the linked line, which should be eliminated) my brain was addled by doing infrastructure work in Python; that's my excuse anyway.

Reply:
What should the line be instead? I can create a PR

Reply:
Ugh, never mind; I was wrong about that, too! 😭 
dave.credibility = 0️⃣